crates/markdown_preview/src/markdown_parser.rs

```diff
@@ -642,6 +653,56 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_nested_bold_strikethrough_text() {
+        let parsed = parse("Some **bo~~strikethrough~~ld** text");
+
+        assert_eq!(parsed.children.len(), 1);
+        assert_eq!(
+            parsed.children[0],
+            ParsedMarkdownElement::Paragraph(ParsedMarkdownText {
+                source_range: 0..35,
+                contents: "Some bostrikethroughld text".to_string(),
+                highlights: Vec::new(),
+                region_ranges: Vec::new(),
+                regions: Vec::new(),
+            })
+        );
+
+        let paragraph = if let ParsedMarkdownElement::Paragraph(text) = &parsed.children[0] {
```

Comment:
Could not rely on `assert_eq` above because there is a impl for `PartialEq` on `ParsedMarkdownText` in line 1166 that only asserts source_range and content to be equal. Removing this would greatly complicate the other tests, so I left it as is.